### PR TITLE
Update add item in developers guide

### LINKED ISSUE #112: Update add item implementation in DG

### PR DESCRIPTION
**Pull request by Jiacheng-y**
This is a preliminary update of the implementations section of the developer's guide for add items. 

This update does **not** include any diagrams. Diagrams that need to be added include: 

- sequence diagram for add item
- activity diagram for add item

#112

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -154,6 +154,42 @@ Classes used by multiple components are in the `tracko.commons` package.
...
+
+#### Implementation
+
+The add item command will be executed by `AddItemCommand`. Items added will be stored in `ItemsList`. 
+
+Given below is an example usage scenario and how the add item mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. The `TrackO` will be initialized with the initial TrackO state, and the `ItemsList` will contain sample data.
+
+Step 2. The user executes `addi i/keys q/10` command to add 10 keys to item list in TrackO. The `addi` command creates an `AddItemCommandParser` which checks the necessary input arguments for item name (prefixed by `i/`) and quantity (prefixed by `q/`) are present before parsing the arguments into a `AddItemCommand` object. The `AddItemCommand` calls `Model#addItem()` to add the item and its corresponding quantity into the items list.
```
Comment (melissaharijanto) [inline docs/DeveloperGuide.md]: Very minor issue here, but considering that you used 'an' for `AddItemCommandParser`, I think 'an' should also be used for `AddItemCommand`.

**Review by melissaharijanto — APPROVED** [review]
Looks good to me!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Jiacheng-y) [reply docs/DeveloperGuide.md]: Thank you for catching it! I've made the change!

**Review by janelleljt — APPROVED** [review]
Approved for now but should definitely update with most recent functionality in the near future :"

**Pull request merged**